server/controllers/Conversations.js

```diff
@@ -15,125 +15,75 @@ module.exports.addConversation = async (req, res) => {
         return;
     }
 
-    // create the conversation with the input attributes
-    let conversation = await Conversation.create({
-        title: req.body.title,
-        imageUrl: req.body.imageUrl,
-        // if the options were null set their values to false else use the input values
-        userEditableImage: req.body.userEditableImage==null ? false :  req.body.userEditableImage,
-        userEditableTitle: req.body.userEditableTitle==null ? false :  req.body.userEditableTitle,
-    }).catch(err => { // catch any errors
-        res.status(500).send({
-            msg: err.message || "Some error occurred while creating the Conversation."
+    // Managed Transaction
+    let t = await db.sequelize.transaction();
+
+    try {
+        //create the conversation with the input attributes
+        let conversationInfo = await Conversation.create({
+            title: req.body.title,
+            imageUrl: req.body.imageUrl,
+            // if the options were null set their values to false else use the input values
+            userEditableImage: req.body.userEditableImage==null ? false :  req.body.userEditableImage,
+            userEditableTitle: req.body.userEditableTitle==null ? false :  req.body.userEditableTitle,
+        }, { transaction: t });
+
+        // add user creating convo as admin
+        await Participant.create({
+            userId: userId,
+            conversationId: conversationInfo.id,
+            isAdmin: true
+        }, { transaction: t });
+
+        // the participants' userIds will be passed in an array
+        let participants = req.body.participants;
+
+        // add all other users in the participants array as participants in the conversation
+        let isAdmin = participants.length == 1;
+
+        const promises = participants.map(async participant => {
```

Comment:
If client does not pass a participants list (not meaning an empty array) this would break. A check similar to the check done for userId earlier (whether it was null/undefined), should also be done for participants.